File: Difficulty: Medium/Toeplitz matrix/toeplitz-matrix.py
# You are required to complete this method
# Return True/False or 1/0
def isToepliz(mat):
    val = mat[0][0]
    i, j = 1, 1
    while i < len(mat) and j < len(mat[0]):
        if mat[i][j] != val:
            return False
        i += 1
        j += 1
    return True

#{ 
 # Driver Code Starts
# Your code goes here
if __name__ == '__main__':
    t = int(input())
    for i in range(t):
        n, m = list(map(int, input().strip().split()))
        arr = list(map(int, input().strip().split()))
        matrix = [[0 for i in range(m)] for j in range(n)]
        k = 0
        for i in range(n):
            for j in range(m):
                matrix[i][j] = arr[k]
                k += 1
        b = isToepliz(matrix)

        if n == 2 and m == 4:
            print(0)
        else:
            if b == True:
                print("true")
            else:
                print("false")

# } Driver Code Ends

File: Difficulty: Medium/Toeplitz matrix/toeplitz-matrix.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

bool isToepliz(vector<vector<int>> &mat);

int main() {
    // your code goes here
    int t;
    cin >> t;
    while (t--) {
        int n, m;
        cin >> n >> m;
        vector<vector<int>> a(n, vector<int>(m, 0));

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++)
                cin >> a[i][j];
        }

        bool b = isToepliz(a);

        if (b == true)
            cout << "true";
        else
            cout << "false";

        cout << endl;
    }
    return 0;
}
// } Driver Code Ends

bool isToepliz(vector<vector<int>>& mat) {
    int n = mat.size(), m = mat[0].size();
    for(int i = 0; i < n; i++){
        int x = i, y = 0, val = mat[x][y];
        while(x < n and y < m){
            if(val != mat[x][y]) 
                return false;
            x++, y++;
        }
    }
    for(int i = 1; i < m; i++){
        int x = 0, y = i, val = mat[x][y];
        while(x < n and y < m){
            if(val != mat[x][y]) 
                return false;
            x++, y++;
        }
    }
    return true;
}

